sdks/python/apache_beam/runners/pipeline_context.py

```diff
@@ -118,13 +120,19 @@ def get_by_id(self, id):
 
   def get_by_proto(self, maybe_new_proto, label=None, deduplicate=False):
     # type: (message.Message, Optional[str], bool) -> str
+    obj = self._obj_type.from_runner_api(
```

Comment:
I'm not sure this is always safe (e.g. for cross language). Why was this needed?

Reply:
With the bug, there was duplication of environments with the same key, and meaningless label 'None'

Reply:
Did you mean https://github.com/apache/beam/pull/12067? The placement of label here does look like a bug. It still makes me nervous to unconditionally call from_runner_api here (from an xlang perspective) just to assign an id to a proto. 

@TheNeuralBit any thoughts on how best to resolve this?

Reply:
Hm I'll need to spend my time to jog my memory on this. I think I may have naively propagated some existing logic, prior to #12067 the return statement was:
```
return self.put_proto(self._unique_ref(label), maybe_new_proto)
```
And the first argument of `unique_ref` is `obj`. I vaguely recall puzzling over this for a while.

Reply:
Passing label to `unique_ref`'s obj arg goes back as long as `get_by_proto` has existed. See https://github.com/apache/beam/pull/7729

Reply:
Shouldn't we just require these environments to not have a meaningless `label`?

Reply:
(Didn't see last comments and yes, the bug was before.)

Reply:
It seems like `get_by_proto` has no usage outside of tests and this PR.
I can look into deleting this method altogether and find another solution for my purposes. Alternatively, I can assert that the input is an Environment for now and/or some 'allowed' entities (Coders?) where XLang concerns don't apply

Reply:
Let's add a todo here--if it causes issues at least we'll have this recorded for posterity and hopefully we can clean things up once non-portability is gone.

Reply:
Added and assertion as discussed offline and a TODO.